### ISSUE TITLE
`Layout/ExtraSpacing` and `Layout/SpaceBeforeFirstArg` does not work as expected

### ISSUE THREAD

**Issue by ytkg** [rubocop/rubocop#10378]
## Expected behavior

The following code registers an offense about `Layout/ExtraSpacing` and `Layout/SpaceBeforeFirstArg`.

```ruby
$ cat Gemfile
# frozen_string_literal: true

source 'https://rubygems.org'

gemspec

gem  'rake'
gem  'rspec'
gem 'rubocop'
```

## Actual behavior

```bash
$ bundle exec rubocop --debug Gemfile
For /Users/ytkg/Workspace/nature_remo_e: configuration from /Users/ytkg/Workspace/nature_remo_e/.rubocop.yml
Default configuration from /Users/ytkg/.rbenv/versions/3.1.0/lib/ruby/gems/3.1.0/gems/rubocop-1.25.0/config/default.yml
Use parallel by default.
Running parallel inspection
Loading cache from /Users/ytkg/.cache/rubocop_cache/6101d0ce49f8ad59cfbc9557f2481ffde4e97306/6d7a3b621ca1730e04accd938619e4bdab66cfb1/b4c06b8b06af8c87c4c7b71e75d0ec11af94397b
Inspecting 1 file
Scanning /Users/ytkg/Workspace/nature_remo_e/Gemfile
Loading cache from /Users/ytkg/.cache/rubocop_cache/6101d0ce49f8ad59cfbc9557f2481ffde4e97306/6d7a3b621ca1730e04accd938619e4bdab66cfb1/b4c06b8b06af8c87c4c7b71e75d0ec11af94397b
.

1 file inspected, no offenses detected
Finished in 0.18543399963527918 seconds
```

## Steps to reproduce the problem

```ruby
$ cat Gemfile
# frozen_string_literal: true

source 'https://rubygems.org'

gemspec

gem  'rake'
gem  'rspec'
gem 'rubocop'
```

```bash
$ bundle exec rubocop Gemfile
Inspecting 1 file
.

1 file inspected, no offenses detected

Tip: Based on detected gems, the following RuboCop extension libraries might be helpful:
  * rubocop-rake (https://rubygems.org/gems/rubocop-rake)
  * rubocop-rspec (https://rubygems.org/gems/rubocop-rspec)

You can opt out of this message by adding the following to your config (see https://docs.rubocop.org/rubocop/extensions.html#extension-suggestions for more options):
  AllCops:
    SuggestExtensions: false
```

### And

For some reason, `webmock` registers an offense.

```ruby
$ cat Gemfile
# frozen_string_literal: true

source 'https://rubygems.org'

gemspec

gem  'rake'
gem  'rspec'
gem 'rubocop'
gem  'webmock'
```

```
$ bundle exec rubocop Gemfile
Inspecting 1 file
C

Offenses:

Gemfile:11:4: C: [Correctable] Layout/ExtraSpacing: Unnecessary spacing detected.
gem  'webmock'
   ^
Gemfile:11:4: C: [Correctable] Layout/SpaceBeforeFirstArg: Put one space between the method name and the first argument.
gem  'webmock'
   ^^

1 file inspected, 2 offenses detected, 2 offenses auto-correctable

Tip: Based on detected gems, the following RuboCop extension libraries might be helpful:
  * rubocop-rake (https://rubygems.org/gems/rubocop-rake)
  * rubocop-rspec (https://rubygems.org/gems/rubocop-rspec)

You can opt out of this message by adding the following to your config (see https://docs.rubocop.org/rubocop/extensions.html#extension-suggestions for more options):
  AllCops:
    SuggestExtensions: false
```

## RuboCop version

```bash
$ bundle exec rubocop -V
1.25.0 (using Parser 3.1.0.0, rubocop-ast 1.15.1, running on ruby 3.1.0 x86_64-darwin20)
```

**Comment by ytkg** [comment]
I understood. Thank you!

**Comment by jonas054** [comment]
I agree with @issyl0's observations. And it _is_ intentional, not a bug. So... closing.
